shell/platform/android/io/flutter/embedding/FlutterEngine.java

```diff
@@ -0,0 +1,262 @@
+package io.flutter.embedding;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.util.Log;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.flutter.embedding.legacy.FlutterPluginRegistry;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.view.FlutterRunArguments;
+
+/**
+ * A single Flutter execution environment.
+ *
+ * A {@code FlutterEngine} can execute in the background, or it can be rendered to the screen by
+ * using the accompanying {@link FlutterRenderer}.  Rendering can be started and stopped, thus
+ * allowing a {@code FlutterEngine} to move from UI interaction to data-only processing and then
+ * back to UI interaction.
+ *
+ * To start running Flutter within this {@code FlutterEngine}, use {@link #runFromBundle(FlutterRunArguments)}.
+ * The {@link #runFromBundle(FlutterRunArguments)} method may not be invoked twice on the same
+ * {@code FlutterEngine}.
+ *
+ * To start rendering Flutter content to the screen, use {@link #getRenderer()} to obtain a
+ * {@link FlutterRenderer} and then attach a {@link FlutterRenderer.RenderSurface}.  Consider using
+ * a {@link FlutterView} as a {@link FlutterRenderer.RenderSurface}.
+ * TODO(mattcarroll): for the above instructions for RenderSurface to be true, we need to refactor
+ *                    the native relationships and also the PlatformViewsController relationship.
+ */
+public class FlutterEngine implements BinaryMessenger {
+  private static final String TAG = "FlutterEngine";
+
+  private final Resources resources;
+  private final FlutterJNI flutterJNI;
+  private final FlutterRenderer renderer;
+  private final FlutterPluginRegistry pluginRegistry;
+  private final Map<String, BinaryMessageHandler> mMessageHandlers;
+  private final Map<Integer, BinaryReply> mPendingReplies = new HashMap<>();
+  private long nativeObjectReference;
+  private boolean isBackgroundView; // TODO(mattcarroll): rename to something without "view"
+  private boolean applicationIsRunning;
+  private int mNextReplyId = 1;
+
+  FlutterEngine(
+      Context context,
```

Comment:
We're bringing it in specifically for the PluginRegistry, so the answer is in there somewhere.  I'm guessing it's actually cast to an Activity.  I will be re-architecting the PluginRegistry so that it does not require an Activity, and when I do that I will probably be able to eliminate the Context from this constructor all together.